test/utils_test.cpp

```diff
@@ -140,4 +140,13 @@ TEST(UtilTestsGroup, Range_test1) {
     v1 = Range(5,7);
     CHECK_EQUAL(2, v1.size());
     CHECK_EQUAL(5, v1[0]);
+}
+
+TEST(UtilsTestsGroup, intToPaddedString_test)
+{
+    CHECK_EQUAL("01", intToPaddedString(1,2));
+    CHECK_EQUAL("0100", intToPaddedString(100,4));
+    CHECK_EQUAL("100", intToPaddedString(100,2));
+    CHECK_EQUAL("1", intToPaddedString(1,1));
+    CHECK_EQUAL("0000056", intToPaddedString(56,7));

```

Comment:
can you add the corner case number=0 and a negative example to the tests?

Reply:
negative example behaviour is undefined so probably shouldn't enforce a test on that. Can do 0 edge case.

Reply:
me (personally) I would add an assert(x>=0) if the behavior is undefined; I would prefer to know where/why things went wrong. Up to you; not sure if it fits the project style.